lighthouse-core/config/config.js

```diff
@@ -381,9 +381,13 @@ class Config {
    * @param {!Array<string>=} skipAuditIds
    * @return {!Object<string, {audits: !Array<{id: string}>}>}
    */
-  static filterCategoriesAndAudits(oldCategories, categoryIds = [], auditIds = [],
-      skipAuditIds = []) {
+  static filterCategoriesAndAudits(oldCategories, categoryIds, auditIds, skipAuditIds) {
     const categories = {};
+    const filterByIncludedCategory = !!categoryIds;
+    const filterByIncludedAudit = !!auditIds;
+    categoryIds = categoryIds || [];
+    auditIds = auditIds || [];
```

Comment:
Maybe we should drop the default `[]` for the whitelisted audits and categories, since there's a difference between `undefined` (whitelist everything) and `[]` (whitelist nothing). It'll require a few more conditionals below, but it may make things easier to read/reason about